File: fsharp/grains/Grains.fs
module Grains

let square (n: int): Result<uint64,string> = 
    match n with
    | _ when n < 1 -> Error "square must be between 1 and 64"
    | n -> Ok (pown 2UL (n - 1))

let squareOrZero (n: int): uint64 =
    match square n with
    | Ok x -> x
    | Error _ -> 0UL

let total: Result<uint64,string> =
    seq { 1 .. 64 }
    |> Seq.sumBy squareOrZero
    |> Ok

File: python/grains/grains.py
NUM_SQUARES = 64

def square(number):
    if number < 1 or number > NUM_SQUARES:
        raise ValueError(f"square must be between 1 and {NUM_SQUARES}")

    return 2 ** (number - 1)


def total():
    return sum([square(n) for n in range(1, NUM_SQUARES + 1)])
